internal/services/storage/storage_account_resource.go

```diff
@@ -1246,6 +1247,15 @@ func resourceStorageAccount() *pluginsdk.Resource {
 					}
 				}
 
+				if !v.(*clients.Client).Features.Storage.DataPlaneAvailable {
+					if _, ok := d.GetOk("queue_properties"); ok {
+						return fmt.Errorf("cannot configure 'queue_properties' when the Provider Feature 'data_plane_available' is set to 'false'")
+					}
+					if _, ok := d.GetOk("static_website"); ok {
+						return fmt.Errorf("cannot configure 'static_website' when the Provider Feature 'data_plane_available' is set to 'false'")
+					}
+				}
```

Comment:
Seeing this for a second time, I'm wondering whether we may run into the following issue:

1. User has `static_website` and or `queue_properties` set
2. User sets `data_plane_available = false` and removes `static_website` and `queue_properties`
3. TF fails to mark the removal of the blocks as a diff because they're both O+C and we incorrectly error

Reply:
This is explicitly to catch point 1 and error at plan time, and the feature is documented to this effect. WRT 2 and 3, I think this is an unlikely condition since the feature applies to creation of accounts? If these properties are already set / the resource already exists, there would be no point disabling the data plane access?

Reply:
So not a valid scenario